src/components/Select/Select.tsx

```diff
@@ -180,6 +180,12 @@ export const Select: FC<SelectProps> = React.forwardRef(
     // Update options on change
     useEffect(() => {
       onOptionsChange?.(getSelectedOptionValues(), getSelectedOptions());
+
+      // Determine first render to help determine the Select interaction is intentional
+      if (firstRender.current) {
+        firstRender.current = false;
+        return;
+      }
       if (multiple) {
```

Comment:
Honestly, this is still worrying me a bit. Our intent here is to have this element get focus specifically when a user sets a value on it, but not when a value is programmatically set right? This check prevents the focus from getting set on the first focus, but I'm wondering what other cases could trigger focus

What if we do an immediate render while making an async call to get a list of available options which we then set a default value for?

Other possibility would be a form where someone sets a value on one property, which then triggers a different value on a select.

Can we do something here to be more explicit about doing this only when we _know_ it's been initiated by a user?

Reply:
What are your thoughts?

Reply:
Technically there is no way to explicitly determine a user has initiated the on change event of the input, as this may be done programmatically. The above code only ever triggers when the input onchange event occurs. I'll investigate a different approach.

Reply:
Added an additional compare for the previous value of `dropdownVisible` to improve the heuristics of the code.